diff -urNp strongswan-5.2.0-patched/src/libstrongswan/plugins/plugin_loader.c strongswan-5.2.0-current/src/libstrongswan/plugins/plugin_loader.c
--- strongswan-5.2.0-patched/src/libstrongswan/plugins/plugin_loader.c	2014-07-18 15:24:06.260406537 -0400
+++ strongswan-5.2.0-current/src/libstrongswan/plugins/plugin_loader.c	2014-07-18 15:30:28.672243729 -0400
@@ -356,7 +356,7 @@ static plugin_entry_t *load_plugin(priva
 			return NULL;
 		}
 	}
-	handle = dlopen(file, RTLD_LAZY);
+	handle = dlopen(file, RTLD_NOW|RTLD_GLOBAL);
 	if (handle == NULL)
 	{
 		DBG1(DBG_LIB, "plugin '%s' failed to load: %s", name, dlerror());
